src/gui/plugins/visualize_lidar/VisualizeLidar.cc

```diff
@@ -0,0 +1,445 @@
+    public: rendering::ScenePtr scene;
+
+    /// \brief Pointer to LidarVisual
+    public: rendering::LidarVisualPtr lidar;
+
+    /// \brief Visual type for lidar visual
+    public: rendering::LidarVisualType visualType;
+
+    /// \brief URI sequence to the lidar link
+    public: std::string lidarString{""};
+
+    /// \brief LaserScan message from sensor
+    public: msgs::LaserScan msg;
+
+    /// \brief Pose of the lidar visual
+    public: math::Pose3d lidarPose{math::Pose3d::Zero};
+
+    /// \brief Current state of the checkbox
+    public: bool checkboxState{false};
+
+    /// \brief Topic name to subscribe
+    public: std::string topicName;
+
+    /// \brief Entity representing the sensor in the world
+    public: gazebo::Entity lidarEntity;
+
+    /// \brief Minimum range for the visual
+    public: double minVisualRange;
+
+    /// \brief Maximum range for the visual
+    public: double maxVisualRange;
+
+    /// \brief Mutex for variable mutated by the checkbox and spinboxes
+    /// callbacks.
+    /// The variables are: msg, visualType, minVisualRange and
+    /// maxVisualRange
+    public: std::mutex serviceMutex;
+
+    /// \brief Initialization flag
+    public: bool initialized{false};
+
+    /// \brief lidar visual display dirty flag
+    public: bool visualDirty{false};
+
+    /// \brief lidar sensor entity dirty flag
+    public: bool lidarEntityDirty{false};
+
+    /// \brief Name of the world
+    public: std::string worldName;
+  };
+}
+}
+}
+
+using namespace ignition;
+using namespace gazebo;
+
+/////////////////////////////////////////////////
+VisualizeLidar::VisualizeLidar()
+  : GuiSystem(), dataPtr(new VisualizeLidarPrivate)
+{
+  // no ops
+}
+
+/////////////////////////////////////////////////
+VisualizeLidar::~VisualizeLidar() = default;
+
+/////////////////////////////////////////////////
+void VisualizeLidar::LoadLidar()
+{
+  auto loadedEngNames = rendering::loadedEngines();
+  if (loadedEngNames.empty())
+    return;
+
+  // assume there is only one engine loaded
+  auto engineName = loadedEngNames[0];
+  if (loadedEngNames.size() > 1)
+  {
+    igndbg << "More than one engine is available. "
+      << "VisualizeLidar plugin will use engine ["
+        << engineName << "]" << std::endl;
+  }
+  auto engine = rendering::engine(engineName);
+  if (!engine)
+  {
+    ignerr << "Internal error: failed to load engine [" << engineName
+      << "]. VisualizeLidar plugin won't work." << std::endl;
+    return;
+  }
+
+  if (engine->SceneCount() == 0)
+    return;
+
+  // assume there is only one scene
+  // load scene
+  auto scene = engine->SceneByIndex(0);
+  if (!scene)
+  {
+    ignerr << "Internal error: scene is null." << std::endl;
+    return;
+  }
+
+  if (!scene->IsInitialized() || scene->VisualCount() == 0)
+  {
+    return;
+  }
+
+  // Create lidar visual
+  igndbg << "Creating lidar visual" << std::endl;
+
+  auto root = scene->RootVisual();
+  this->dataPtr->lidar = scene->CreateLidarVisual();
```

Comment:
I don't see the visual destroyed? we need to be careful not to leave any dangling visuals around when the plugin exits

Reply:
following up on this, we should destroy the visual when the user closes the plugin, i.e.  by clicking on `X` in the corner of the plugin window. Currently I see that lidar visual remains in the scene after exiting the plugin, and when I open the same plugin again, a duplicate lidar visual gets created.